zuul-core/src/main/java/com/netflix/zuul/filters/endpoint/ProxyEndpoint.java

```diff
@@ -389,11 +378,14 @@ private void storeAndLogOriginRequestInfo() {
         if (attemptToChosenHostMap == null) {
             attemptToChosenHostMap = new HashMap<>();
         }
-        String ipAddr = origin.getIpAddrFromServer(chosenServer.get());
-        if (ipAddr != null) {
-            attemptToIpAddressMap.put(attemptNum, ipAddr);
-            eventProps.put(CommonContextKeys.ZUUL_ORIGIN_ATTEMPT_IPADDR_MAP_KEY, attemptToIpAddressMap);
-            context.put(CommonContextKeys.ZUUL_ORIGIN_ATTEMPT_IPADDR_MAP_KEY, attemptToIpAddressMap);
+
+        if (chosenServer.get() != null) {
```

Comment:
You probably want to check for the sentinel value of `DiscoveryResult.EMPTY`. This is unlikely to be null, as part of resolver lookup.

Reply:
This is purely a null check to prevent NPE.

Reply:
Okay. Probably worth adding a comment for when that's likely to happen.